### PR TITLE
Added a splash screen on start-up (re-done)

### PR DESCRIPTION
**Pull request by BenjaminAmos**
# Description
This pull request implements a splash screen that is shown whilst the game is loading its modules on start-up.

# Testing
- Start the game
- The Destination Sol logo should be visible in the centre of the screen. It should be replaced with the game window once the game has finished loading its modules.

# Notes
- The modules are initialised in another thread, although no synchronisation statements are in place. I would not assume this to be a problem though, as the spawned thread only writes to variables that the main thread reads from.
- You will have to regenerate the project files ("gradlew idea" or "gradlew eclipse") for these changes to work.
- The Eclipse changes worked for me but, as I do not use Eclipse usually for development, they may not work for others.
- This is the supersedes #358.

### CODE REVIEW

File: engine/src/test/java/org/destinationsol/testingUtilities/InitializationUtilities.java [hunk 1]
```diff
@@ -33,7 +34,7 @@ public static void init() {
         }
         initialized = true;
         DebugOptions.DEV_ROOT_PATH = "engine/src/main/resources/";
-        final LwjglApplication application = new LwjglApplication(new SolApplication());
+        final LwjglApplication application = new LwjglApplication(new SolApplication(new ModuleManager()));
         try {
             Thread.sleep(10000L); // Magic happens here. If too much tests fail for you, increase this number.
```
Comment (Adrijaned) [inline engine/src/test/java/org/destinationsol/testingUtilities/InitializationUtilities.java]: About the "Magic" - it was just a quick and dirty way to assure modules are loaded and game is initialized before running the tests - could you check if this is still needed and/or can be easily replaced? (Maybe moving the module initialization fixed this)

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by Adrijaned** [comment]
Great! Just please try to avoid force-pushing unless necessary :slightly_smiling_face:

File: engine/src/test/java/org/destinationsol/testingUtilities/InitializationUtilities.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (BenjaminAmos) [reply engine/src/test/java/org/destinationsol/testingUtilities/InitializationUtilities.java]: I actually partially broke the tests with my previous commit. The engine build.gradle changes prevented the resources from being copied into the output directory. I have fixed that now.

The work-around does not appear to be needed anymore, so I have removed it for now.

**Pull request merged**